content/how-to-use-supertokens-custom-ui-with-vuejs/index.md

```diff
@@ -0,0 +1,608 @@
+
+![invalid-email-error](./invalid-email-error.png)
+
+Similarly, we have the `signUp` method where we invoke the `emailPasswordSignUp` function from `supertokens-web-js` to handle the signup flow. 
+
+For social login, we’re using Google and Github Authentication providers. When the `onGithubPressed` or `onGooglePressed` functions are called, we call the `getAuthorisationURLWithQueryParamsAndSetState` method and pass the provider name in the parameters. We also provide a callback URL as `authorisationURL` parameter that each provider will redirect to after the authentication process is completed. In our example, we use http://localhost:3000/auth/callback/google for Google and [http://localhost:3000/auth/callback/github for GitHub . Note that this URL needs to be configured on the provider’s dashbaord.
+
+Here are the functions for Github and Google respectively:
+
+```vue
+...
+      onGithubPressed: async function () {
+          const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
+              providerId: "github",
+
+              // This is where github should redirect the user back after login or error.
+              // This URL goes on the github dashboard as well.
+              authorisationURL: `http://localhost:3000/auth/callback/github`,
+          });
+
+          window.location.assign(authUrl);
+      },
+      onGooglePressed: async function () {
+          const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
+              providerId: "google",
+
+              // This is where google should redirect the user back after login or error.
+              // This URL goes on the google dashboard as well.
+              authorisationURL: `${websiteDomain}/auth/callback/google`,
+          });
+
+          window.location.assign(authUrl);
+      },
+ ……
+```
+
+The `getAuthorisationURLWithQueryParamsAndSetState` method returns the callback URL that takes us to the `/auth/callback` page that we discussed above.
+
+After social authentication is completed and the user is redirected to the `/auth/callback/<provider>` route, we call the `thirdPartySignInAndUp` function from `supertokens-web-js` which consumes the authorisation code to sign in the user.
+
+Here is the function that handles the above flow in the `AuthCallbackView` component inside `/src/views/AuthCallbackView.vue` file:
+
+```vue
+  mounted: async function () {
+       try {
+      
+           const response = await ThirdPartyEmailPassword.thirdPartySignInAndUp({});
+           if (response.status !== "OK") {
+               return window.location.assign("/auth?error=signin");
+           }
+           // sign in successful. The session tokens are handled
+           // automatically via our SDK.
+           window.location.assign("/");
+       } catch (_) {
+              window.location.assign("/auth?error=signin");
+       }
+   }
+
+```
```

Comment:
```suggestion
```ts
mounted: async function () {
     try {
    
         const response = await ThirdPartyEmailPassword.thirdPartySignInAndUp();
         if (response.status !== "OK") {
         // either the user did not complete the login process, or something else went wrong.
             return window.location.assign("/auth?error=signin");
         }
         
         // sign in successful.
         // The session tokens are handled automatically via our SDK.
         window.location.assign("/");
         
     } catch (_) {
            window.location.assign("/auth?error=signin");
     }
}
```
```

Reply:
All the code snippets need proper formatting. The indentations throughout this blog are inconsistent.